### ISSUE TITLE
Does not work with vue 2.4.x

### ISSUE THREAD

**Issue by michaelchiche** [algolia/vue-instantsearch#186]
**Do you want to request a *feature* or report a *bug*?**
I report a bug.

As in the link shown: https://codepen.io/anon/pen/KqLMXY#anon-login

**Bug: What is the current behavior?**
There is an exception which breaks the whole app

**Bug: What is the expected behavior?**
It should work fine

**Bug: What browsers are impacted? Which versions?**
Tested on latest versions of chrome, safari & firefox

**Bug: What is the proposed solution?**
Lock on 2.3.x for now or upgrade code ?

From what I can understand, it looks like the DI does not work properly with this upgrade

**What is the version you are using? Always use the latest one before opening a bug issue.**
I did a fresh npm install following the guide here: vue-instantsearch.netlify.com

**Comment by Haroenv** [comment]
Hey @michaelchiche, we found out about this change yesterday as well, the problem is indeed that the dependency injection -- the context -- changed from Vue 2.3 to 2.4. We didn't find an immediate solution to this, but definitely will keep an eye on this and have it fixed asap. For now you can still use 2.3.X indeed

see also https://github.com/vuejs/vue/issues/6093

**Comment by Haroenv** [comment]
A workaround in that thread is to do this after your import of Vue and before your import of Vue InstantSearch:

```
Vue.config.optionMergeStrategies.inject = (parent, child) => Vue.util.extend(Vue.util.extend({}, parent), child)
```

**Comment by rayrutjes** [comment]
A fix has been merged. Now waiting for release of 2.4.2 🎉

**Comment by rayrutjes** [comment]
@michaelchiche should now be fixed in 2.4.2 of Vue.js
Please re-open the issue if the new release does not solve your initial problem.
